Msi/source management ui

larsroettig opened the pull request:
<!--- Provide a general summary of the Pull Request in the Title above -->

### Description
<!--- Provide a description of the changes proposed in the pull request -->

### Fixed Issues (if relevant)
<!--- Provide a list of fixed issues in the format magento/magento2#<issue_number>, if relevant  -->
1. magento/magento2#<issue_number>: Issue title
2. ...

### Manual testing scenarios
<!--- Provide a set of unambiguous steps to test the proposed code change -->
1. ...
2. ...

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

Commit:
add middle name to checkout address html templates #8878

Commit:
Make deployed_version.txt editor friendly

In production environments with a CDN for static files it comes in handy that you can force a browser cache refresh by changing the number manually without having to go into maintenance for a full deploy.

If you edit the `pub/static/deployed_version.txt` with vim, nano or `echo '123456789' > pub/static/deployed_version.txt` you can unintentionally create a newline at the end of the file.
This will result in:

```html
    var BASE_URL = 'https://magento2.dev/sitemanager/admin/index/index/key/ec38dd3ded42b71db14166d6bcdfc56d4e7d4b7801b9808373ae4b3563b65513/';
    var FORM_KEY = 'dKArsQU6NhimXr6Z';
    var require = {
        "baseUrl": "https://magento2.dev/pub/static/version1492766286
/adminhtml/Magento/backend/en_US"
    };
```

With that baseUrl all resources using baseUrl will return the 404 page.

Adding a simple `trim()` here will remove that risk.

omiroshnichenko commented on app/code/Magento/Inventory/view/adminhtml/ui_component/inventory_source_form.xml:
```diff
@@ -0,0 +1,89 @@
+            </item>
+        </argument>
+        <field name="source_id">
+            <argument name="data" xsi:type="array">
```
Use also settings, instead of data -> config.

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/AbstractButton.php:
```diff
@@ -0,0 +1,76 @@
+/**
+ * Class AbstractButton
+ */
+abstract class AbstractButton
```
Try to avoid usage of abstract classes in such cases.
We do not encourage usage and introduction of the new inheritance based APIs

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/AbstractButton.php:
```diff
@@ -0,0 +1,76 @@
+     *
+     * @return bool
+     */
+    protected function isEditMode()
```
Please avoid using protected methods

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/AbstractButton.php:
```diff
@@ -0,0 +1,76 @@
+     *
+     * @return null | int
+     */
+    protected function getSourceId()
```
Please avoid using protected methods

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/AbstractButton.php:
```diff
@@ -0,0 +1,76 @@
+        try {
+            $sourceId = $this->context->getRequest()->getParam('id');
+            return $this->sourceRepository->getById($sourceId)->getId();
+        } catch (NoSuchEntityException $exception) {
```
Exception must be processed

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/Delete.php:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * @return string
+     */
+    private function getDeleteUrl()
```
This method is not really needed, as it is used only once. You may inline this code in this case.

ishakhsuvarov commented on app/code/Magento/Inventory/Controller/Adminhtml/Sources/NewAction.php:
```diff
@@ -0,0 +1,41 @@
+    /**
+     * @var \Magento\Backend\Model\View\Result\ForwardFactory
+     */
+    protected $resultForwardFactory;
```
Magento does not encourage inheritance-based API. This method should be private.

ishakhsuvarov commented on app/code/Magento/Inventory/Controller/Adminhtml/Sources/Save.php:
```diff
@@ -0,0 +1,40 @@
+    /**
+     * @var \Magento\Framework\View\Result\PageFactory
+     */
+    protected $resultPageFactory;
```
Magento does not encourage inheritance-based API. This property should be private.

ishakhsuvarov reviewed (changes requested):
* Term `Source` should be used instead of `Warehouse`
* UI Component declarations should stick to the up-to-date format (please see integration test failures)

ishakhsuvarov commented on app/code/Magento/Inventory/Block/Adminhtml/Edit/Button/Delete.php:
```diff
@@ -0,0 +1,78 @@
+    public function getButtonData()
+    {
+        $data = [];
+        // This part should be refactored. Implemented for test purposes.
```
* There is no reason to get Source by Id from SourceRepository, since delete controller will validate it anyways.
* We should research building delete url on frontend, to avoid getting Request in block, which is very bad.

slamking committed:
MAGETWO-69185: Unable to create shipping label if Region not specified for shipping origin

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-62271

# Conflicts:
#	dev/tests/functional/tests/app/Magento/CatalogInventoryConfigurableProduct/Test/TestCase/CreateConfigurableProductEntityTest.xml

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
Merge branch 'mni/source-management' into mni/source-management-ui

Commit:
Merge remote-tracking branch 'mainline/develop' into mni/source-management-ui

# Conflicts:
#	composer.lock

slamking committed:
Merge remote-tracking branch 'origin/develop' into MAGETWO-69185

slopukhov committed:
MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories

Commit:
Merge remote-tracking branch 'origin/MAGETWO-54702' into PR_08062017

Commit:
Merge remote-tracking branch 'origin/MAGETWO-59514' into PR_08062017

Commit:
Merge remote-tracking branch 'origin/MAGETWO-61189' into PR_08062017

Commit:
Merge remote-tracking branch 'origin/MAGETWO-64952' into PR_08062017

Commit:
Merge remote-tracking branch 'origin/MAGETWO-66885' into PR_08062017

igrybkov committed:
Merge pull request #1164 from magento-dragons/MAGETWO-62271

Fixed issues:
- MAGETWO-64249 Issue with integration test Magento/Setup/Console/Command/I18nCollectPhrasesCommandTest.php on Travis CI
- MAGETWO-62271 Inconsistent saving of Stock Data

igrybkov committed:
MAGETWO-62227: Unable to sort attributes table when trying to Add Attribute to a product

Commit:
magento-engcom/magento2#25: Source API Implementation

-- index action

Commit:
magento-engcom/magento2#25: Source API Implementation

omiroshnichenko committed:
Merge branch 'develop' of github.com:magento-engcom/magento2ce into MAGETWO-69690-PR-9690

omiroshnichenko committed:
MAGETWO-69690: Add froogaloop library as a dependency to load-player module #9690

slamking committed:
MAGETWO-69185: Unable to create shipping label if Region not specified for shipping origin

nikshostko committed:
MAGETWO-63054: [Catalog] MSRP field is not displayed for bundle products with fixed price

slamking committed:
MAGETWO-69185: Unable to create shipping label if Region not specified for shipping origin

Commit:
MAGETWO-69749: Fixed issue causing static test failure to report success on Travis #9872

 - Merge Pull Request magento/magento2#9872 from davidalger/magento2:feature/fix-travis-tests

igrybkov committed:
Merge pull request #1168 from magento-dragons/DRAGONS-BUGFIXES

Fixed an issue:
- MAGETWO-62227 Unable to sort attributes table when trying to Add Attribute to a product

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-56796-Any-admin-user-can-upload-Favicon-Icon' into Okapis-develop-pr

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-69015-Limited-Permissions' into Okapis-develop-pr

Commit:
MAGETWO-67438: Exception Thrown with Multiselect Widget Condition

Commit:
MAGETWO-69749: Fixed issue causing static test failure to report success on Travis #9872

Commit:
MAGETWO-69551: Coupon codes not showing in invoice print out #9216 #9780

cpartica committed:
Merge pull request #1177 from magento-okapis/Okapis-develop-pr

[Okapis] P0 P1 bugs

shiftedreality committed:
Merge branch 'develop' of https://github.com/magento/magento2ce into MAGETWO-69556

slamking committed:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69185

woutersamaey committed:
Improved error logging: original exception was not passed as $cause

zakdma committed:
MAGETWO-65468: [FT] Magento\Catalog\Test\TestCase\Product\CreateSimpleProductEntityTest fails on Jenkins

zakdma committed:
Merge remote-tracking branch 'origin/develop' into develop-pr4

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-64221

ishakhsuvarov committed:
MAGETWO-69370: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69370-PR-4450

ishakhsuvarov committed:
MAGETWO-69370: Add ability to use tree-massactions ("sub-menus") on Sales > Orders grid #4450
 - Updated configuration to use the correct format

woutersamaey committed:
Added missing logging when saving a product fails

ishakhsuvarov committed:
MAGETWO-69780: Improved error logging: original exception was not passed as  #9890

 - Merge Pull Request magento/magento2#9890 from woutersamaey/magento2:develop

woutersamaey committed:
Added .DS_Store to .gitignore for Mac users

ishakhsuvarov committed:
MAGETWO-69781: Added .DS_Store to .gitignore for Mac users #9892

 - Merge Pull Request magento/magento2#9892 from woutersamaey/magento2:updated-gitignore

ishakhsuvarov committed:
MAGETWO-69690: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69690-PR-9690

Commit:
MAGETWO-69515: Static files are deployed too slow for multiple locales

ishakhsuvarov committed:
MAGETWO-69692: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69692-PR-9813

serhii-balko committed:
MAGETWO-65468: [FT] Magento\Catalog\Test\TestCase\Product\CreateSimpleProductEntityTest fails on Jenkins

serhii-balko committed:
Merge branch 'MAGETWO-65468' into develop-pr4

kandy committed:
Merge pull request #1173 from magento-falcons/MAGETWO-69556

[Falcons] Delivery of bug fixes for Setup and Deployment

zakdma committed:
Merge remote-tracking branch 'origin/develop' into develop-pr4

ishakhsuvarov committed:
MAGETWO-69737: Merge branch 'patch-2' of https://github.com/redelschaap/magento2 into MAGETWO-69737-PR-7405

Conflicts:
	lib/internal/Magento/Framework/HTTP/Adapter/Curl.php

ishakhsuvarov committed:
MAGETWO-69737: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69737-PR-7405

ishakhsuvarov committed:
MAGETWO-69633: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69633-PR-9368

Commit:
magento-engcom/magento2#25: Source API Implementation

rganin committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67087

Commit:
Merge 760bb5255de6d1c26b40d1410bc5181f054e5c70 into PR_08062017

slopukhov committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67087

slopukhov committed:
MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories

slopukhov committed:
Merge branch 'MAGETWO-67087' of github.com:magento-troll/magento2ce into MAGETWO-67087

dverkade committed:
Added code to the PaymentTokenFactory class for backwards compatibility reasons.

isavchuk-magento committed:
Merge remote-tracking branch 'origin/MAGETWO-60828' into MPI-PR206

isavchuk-magento committed:
Merge remote-tracking branch 'origin/MAGETWO-69340' into MPI-PR206

isavchuk-magento committed:
MAGETWO-69539: PHP "soap" extension is not declared in composer.json but can be used by Magento modules

- Fixed conflict composer.lock

isavchuk-magento committed:
MAGETWO-69539: PHP "soap" extension is not declared in composer.json but can be used by Magento modules

- Updated composer.lock

ishakhsuvarov committed:
MAGETWO-69689: Update CONTRIBUTING.md and README.md of Magento 2 CE repository

ishakhsuvarov committed:
MAGETWO-69781: Added .DS_Store to .gitignore for Mac users #9892

ishakhsuvarov committed:
MAGETWO-69780: Improved error logging: original exception was not passed as $cause #9890

ishakhsuvarov committed:
MAGETWO-69728: Fixes layered navigation options being cached using the wrong store id. #9873

ishakhsuvarov committed:
MAGETWO-69692: Use static:: to support late static bindings in Invoice and Creditmemo #9813

ishakhsuvarov committed:
MAGETWO-69690: Add froogaloop library as a dependency to load-player module #9690

ishakhsuvarov committed:
MAGETWO-69633: Redis sess: fix path for persistent_identifier & compression_threshold #9368

ishakhsuvarov committed:
MAGETWO-69370: Add ability to use tree-massactions ("sub-menus") on Sales Orders grid #4450

ishakhsuvarov committed:
MAGETWO-69689: Merge branch 'code-of-conduct' of github.com:magento-engcom/magento2ce into develop-prs

Commit:
MAGETWO-67438: Exception Thrown with Multiselect Widget Condition

Commit:
magento-engcom/magento2#25: Source API Implementation

Commit:
magento-engcom/magento2#25: Source API Implementation

Commit:
magento-engcom/magento2#25: Source API Implementation

add InlineEdit controller

ishakhsuvarov committed:
MAGETWO-69728: Fixes layered navigation options being cached using the wrong store id. #9873

 - Backwards compatibility improvements

ishakhsuvarov committed:
MAGETWO-69690: Add froogaloop library as a dependency to load-player module #9690

 - Coding style improvements

ishakhsuvarov committed:
MAGETWO-69667: Fix bug linked product position not updated if product link already exists #9810

 - Merge Pull Request magento/magento2#9810 from jalogut/magento2:fix-import-product-links-update-position

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix composer.json

Commit:
MAGETWO-69723: Email to a Friend feature #9824

 - Merge Pull Request magento/magento2#9824 from WaPoNe/magento2:WaPoNe-patch-1

Commit:
MAGETWO-69723: Email to a Friend feature #9824

 - fixed template configuration format

Commit:
Merge pull request #1184 from magento-engcom/develop-prs

Public Pull Requests:

magento/magento2#9872
magento/magento2#9780
magento/magento2#9892
magento/magento2#9890
magento/magento2#9873
magento/magento2#9813
magento/magento2#9690
magento/magento2#9368
magento/magento2#4450

Commit:
MAGETWO-69805: Return array of blocks as items instead of array of arrays #9157

 - Merge Pull Request magento/magento2#9157 from tkotosz/magento2:patch-1

Commit:
MAGETWO-69666: Return array of pages as items instead of array of arrays #9823

 - fixed unit tests

Commit:
MAGETWO-69805: Return array of blocks as items instead of array of arrays #9157

 - fixed unit tests

Commit:
MAGETWO-67508: Fixed coding standard violations in the Framework\Validator namespace #9251

 - Merge Pull Request magento/magento2#9251 from dverkade/magento2:Fixed_coding_standard_in_the_Framework-Validator

Commit:
MAGETWO-67500: setup:di:compile returns exit code 0 if errors are found #7780

 - Merge Pull Request magento/magento2#7780 from vaimo/magento2:fix/throw-exception-if-compilation-fails

barbazul committed:
Fixed how path is checked in ClassesTest to make it Windows-compatible

barbazul committed:
Convert slashes in path for windows compatibility

Commit:
PATCH-1 Revert "Make deployed_version.txt editor friendly"

This reverts commit 78aba16084f1476af294f8d47529ecb4f64f0576.

Commit:
Add --refresh-content-version-only option to deploy

With this option you can quickly force a browser cache refresh or
a CDN cache refresh without having to generate all static content.

This comes in handy especially when you have made no code changes,
but the CDN failed to update it's cache the last time you deployed
or when you have made a manual change in one of the static files
as a hotfix on your deployment server, awaiting a suitable moment
for deployment to finalize it.

This reopens #9338.

barbazul committed:
Merge branch 'develop' into fix-integrity-tests

barbazul committed:
Line with long regexp exceeded max line length of 120 chars

barbazul committed:
Handling CLI error as a failure when validating composer.json file

barbazul committed:
Using Command output as message which actually provides more information for debugging than just the path

Commit:
Merge remote-tracking branch 'origin/develop' into PR_08062017

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
MAGETWO-69515: Static files are deployed too slow for multiple locales

isavchuk-magento committed:
Merge branch 'develop' of https://github.com/magento-mpi/magento2ce into MPI-PR206

Commit:
Merge remote-tracking branch 'mainline/develop' into mni/source-management-ui

RuslanKostiv1 committed:
Merge branch 'pr36' into NORD-EXT-PR-36

ishakhsuvarov committed:
MAGETWO-69808: Fix composer validation #9905

 - Merge Pull Request magento/magento2#9905 from barbazul/magento2:fix-composer-validation

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
Fix prefix, middle name and suffix were not prefilled in the checkout

Commit:
MAGETWO-69515: Static files are deployed too slow for multiple locales

slopukhov committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67087

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

Commit:
Merge pull request #1183 from magento-nord/NORD-EXT-PR-36

NORD-EXT-PR-36

Commit:
Merge remote-tracking branch 'origin/develop' into PR_08062017

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

MomotenkoNatalia committed:
Merge pull request #1182 from magento-frontend/PR_08062017

Fixed issues:
- MAGETWO-61189: Stored xss using svg images in Favicon
- MAGETWO-54702: Failed ClearAllCompareProductsTest test due to Alert window.
- MAGETWO-66885: Special Characters like % in widget inside of WYSIWYG gives an error #9452
- MAGETWO-59514: Hard coded "tax_region_id" in the \Magento\Tax\Setup\InstallData
- MAGETWO-64952: Admin login does not handle autocomplete feature correctly

slopukhov committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67087

Commit:
MAGETWO-67500: setup:di:compile returns exit code 0 if errors are found #7780

 - fixed false positive failures

Commit:
MAGETWO-69805: Return array of blocks as items instead of array of arrays #9157

Commit:
MAGETWO-69737: Update Curl.php #7405

Commit:
MAGETWO-69723: Email to a Friend feature #9824

Commit:
MAGETWO-69667: Fix bug linked product position not updated if product link already exists #9810

Commit:
MAGETWO-69666: Return array of pages as items instead of array of arrays #9823

Commit:
MAGETWO-67500: setup:di:compile returns exit code 0 if errors are found #7780

dverkade committed:
When an array is supplied, should go straight to the objectmanager create function.

dverkade committed:
Fixed coding standard violation

hostep committed:
Revert "MAGETWO-69728: Fixes layered navigation options being cached using the wrong store id. #9873"

This reverts commit 58b4c7c312395dfe93ca12d8478a2379a104a476.

Commit:
MAGETWO-69842: Fixes small backwards incompatibility issue created in MAGETWO-69728 #9922

 - Merge Pull Request magento/magento2#9922 from hostep/magento2:fix-magetwo-69728
 - Merged commits:
   1. 76b45a71c15975eab37a4c9ff7d2e701096564ef

Commit:
MAGETWO-68976: [Magento Cloud] Incorrect date format with Arabic language locale

 - filter out any additional locale information besides MmDdYy and the separators (‘/‘,’.’,’-‘)

Commit:
Merge pull request #1186 from magento-engcom/develop-prs

Public Pull Requests:
magento/magento2#9157
magento/magento2#7405
magento/magento2#9824
magento/magento2#9810
magento/magento2#9823
magento/magento2#7780

Commit:
MAGETWO-68877: Issue #7988 Typo changed also added comments for each index, getters and setters. #9484

 - fixed merge conflict

Commit:
Merge branch 'develop' into MAGETWO-68877-PR-9484

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-68877-PR-9484

Commit:
MAGETWO-69842: Fixes small backwards incompatibility issue created in MAGETWO-69728 #9922

Commit:
MAGETWO-68877: Issue #7988 Typo changed also added comments for each index, getters and setters. #9484

Commit:
MAGETWO-68976: [Magento Cloud] Incorrect date format with Arabic language locale

 - filter out any additional locale information besides MmDdYy and the separators (‘/‘,’.’,’-‘)

Commit:
Merge pull request #1187 from magento-engcom/develop-prs

Public Pull Requests

magento/magento2#9922
magento/magento2#9484

Commit:
Add unit tests

Commit:
Don't display deploy strategy and execution time when only changeing the version number

Commit:
Fix phpmd test failure, rename method from get to is

Commit:
Shorten DeployStaticContentTest::testDeploy to be within 100 lines

slopukhov committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67087

Commit:
Move prefix and suffix default values to a new PR

Commit:
Fix #9924, prefill prefix and suffix in checkout shipping address

ishakhsuvarov committed:
MAGETWO-69733: add middle name to checkout address html templates #8878 #8879

 - Merge Pull Request magento/magento2#8879 from ajpevers/magento2:issue-8878
 - Merged commits:
   1. f3c7406253938c70742cac308c12729a99f50a1c
   2. 5f1b460249428e6c80910b672d769099d37619c7
   3. 3c632aa90963107c3450b5e67d6cc1aeafb680f2

ishakhsuvarov committed:
MAGETWO-69733: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69733-PR-8879

ishakhsuvarov committed:
MAGETWO-69832: suggestion from #9338: add some command/option to the deploy command to refresh the version #9915

 - Merge Pull Request magento/magento2#9915 from ajpevers/magento2:patch-1
 - Merged commits:
   1. 78aba16084f1476af294f8d47529ecb4f64f0576
   2. 89e006754b6e8811a972e9b70c4a0e1868cfbc5f
   3. 17a44e252ee0ffabf2362cafb3031b7379f142ba
   4. fbc9db37d12889637f1b7cfe5fdfebb0e56133f6
   5. 4499b6f444b87e18950ee9eb8df243e047f6f916
   6. cc8e8fd488d48cc747907e3b44c4644344d33adf
   7. 9de25f59b9de4757e42e26e1479da22668224ff6

Commit:
upgrade tubalmartin/cssmin to 4.1.0

According to https://github.com/tubalmartin/YUI-CSS-compressor-PHP-port/blob/master/README.md#v410-16-may-2017: Performance: 2x times faster than v4.0.0 after code profiling
Also minor enhancements like:

    [target='_blank']       => [target=_blank]
    [type="something"]      => [type=something]
    background: transparent => background: 0 0
    background: none        => background: 0 0
    #f00                    => red
    padding: 20px 20px      => padding 20px
    0.12                    => .12

total min.css size reduced by 2563 bytes (from 1732225 to 1729662)

Commit:
Merge pull request #1185 from magento-troll/MAGETWO-67087

[Troll] MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories

Commit:
Add the composer.lock file

ishakhsuvarov committed:
MAGETWO-69849: upgrade tubalmartin/cssmin to 4.1.0 #9928

 - Merge Pull Request magento/magento2#9928 from ajpevers/magento2:tubalmartin-cssmin-4.1.0
 - Merged commits:
   1. 2523edac2ccc58c0d628ac10c41f7862726955a7
   2. 911c9b0243b51d699ad7dae69062d3db59f47c5a

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-61560' into NORD-BUGFIXES-PR

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-64221' into NORD-BUGFIXES-PR

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-69185' into NORD-BUGFIXES-PR

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix static tests

ishakhsuvarov committed:
MAGETWO-67455: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67455-PR-4891

vgoncharenko committed:
MAGETWO-67431: Functional test \Magento\ConfigurableProduct\Test\TestCase\CreateConfigurableProductEntityTest fails randomly

ishakhsuvarov committed:
MAGETWO-69849: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69849-PR-9928

vgoncharenko committed:
MAGETWO-67431: Functional test \Magento\ConfigurableProduct\Test\TestCase\CreateConfigurableProductEntityTest fails randomly

- unskip test

ishakhsuvarov committed:
MAGETWO-69808: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69808-PR-9905

isavchuk-magento committed:
Merge branch 'develop' of https://github.com/magento/magento2ce into MPI-PR206

Commit:
Simplify the getDefaultValue method in an attempt to pass PHP Mess Detection checks

avoelkl committed:
Fix typo in comment

Commit:
Bring cyclomatic complexity down from 10 to 6

slamking committed:
MAGETWO-69185: Unable to create shipping label if Region not specified for shipping origin

RuslanKostiv1 committed:
Merge remote-tracking branch 'origin/MAGETWO-69185' into NORD-BUGFIXES-PR

viktym committed:
Merge pull request #1189 from magento-mpi/MPI-PR206

MPI-PR206-2

RuslanKostiv1 committed:
Merge remote-tracking branch 'mainline/develop' into NORD-BUGFIXES-PR

Commit:
MAGETWO-69515: Static files are deployed too slow for multiple locales

Commit:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69515

mzeis committed:
MAGETWO-69861: Fix typo in comment #9932

 - Merge Pull Request magento/magento2#9932 from avoelkl/magento2:avoelkl-shipping-test-typo
 - Merged commits:
   1. 7e65ae4b5218c188c749efd5ffda4194ee08571c

Commit:
Bring cyclomatic complexity down from 6 to 4

Commit:
Merge branch 'develop' into exception-when-attribute-does-not-exist

Commit:
MAGETWO-65610: [GitHub][PR] Throw exception when attribute doesn't exitst magento/magento2#7758

 - Merge Pull Request magento/magento2#7758 from AydinHassan/magento2:exception-when-attribute-does-not-exist
 - Merged commits:
   1. ed132d007bb9f059b27a7acdefe1bec53b3d6b52
   2. 328b94eca5930adcd0e18140eed0c24251d7e9ad

Commit:
MAGETWO-69153: Fixed the Inconsistent Gift Options checkbox labels #9421 #9525

 - Merge Pull Request magento/magento2#9525 from vpiyappan/magento2:develop
 - Merged commits:
   1. f68dd921c7e0d064220780fd301d0c06aa6012b8

Commit:
MAGETWO-69861: Fix typo in comment #9932

Commit:
MAGETWO-69808: Fix composer validation #9905

Commit:
MAGETWO-69733: add middle name to checkout address html templates #8878 #8879

Commit:
MAGETWO-69153: Fixed the Inconsistent Gift Options checkbox labels #9421 #9525

Commit:
MAGETWO-67508: Fixed coding standard violations in the Framework\Validator namespace #9251

Commit:
MAGETWO-67455: Remove faulty index subscription #4891

Commit:
MAGETWO-65610: [GitHub][PR] Throw exception when attribute doesn't exitst magento/magento2#7758

Commit:
Merge pull request #1192 from magento-engcom/develop-prs

Public Pull Requests:
magento/magento2#9932
magento/magento2#9905
magento/magento2#8879
magento/magento2#9525
magento/magento2#9251
magento/magento2#4891
magento/magento2#7758

Commit:
MAGETWO-69868: Fix static integrity classes tests in Windows #9902

 - Merge Pull Request magento/magento2#9902 from barbazul/magento2:fix-integrity-tests
 - Merged commits:
   1. e9f65d6bde08f05b72f29691f40df8c47c4855e6
   2. ad2bb5cdff1ab7ce499b5913874ac38ff9429440
   3. df27f9984526d18a8f2ba2961427e46ae2711d12
   4. 2b80b9eb2840ac91525b4b971a154418dc18fe0d

pdohogne-magento committed:
MAGETWO-69648: Adding version check to upgrade script for User module

Commit:
ISSUE-9924 Merge remote-tracking branch 'upstream/develop' into issue-9924

* upstream/develop: (60 commits)
  Fix typo in comment
  Move prefix and suffix default values to a new PR
  MAGETWO-68877: Issue #7988 Typo changed also added comments for each index, getters and setters. #9484
  Revert "MAGETWO-69728: Fixes layered navigation options being cached using the wrong store id. #9873"
  MAGETWO-67500: setup:di:compile returns exit code 0 if errors are found #7780
  Fix prefix, middle name and suffix were not prefilled in the checkout
  add middle name to checkout address html templates #8878
  Using Command output as message which actually provides more information for debugging than just the path
  Handling CLI error as a failure when validating composer.json file
  MAGETWO-69805: Return array of blocks as items instead of array of arrays #9157
  MAGETWO-69666: Return array of pages as items instead of array of arrays #9823
  MAGETWO-69723: Email to a Friend feature #9824
  MAGETWO-69539: PHP "soap" extension is not declared in composer.json but can be used by Magento modules
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  MAGETWO-63054: [Catalog] MSRP field is not displayed for bundle products with fixed price
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  MAGETWO-67087: Saving category deletes url-rewrites for products in anchor categories
  ...

Commit:
By default, show times in admin grids in the store timezone.

fixes #9426

This test basically said, use the column config value, or use FALSE:

    $timezone = isset($this->getConfiguration()['timezone'])
        ? $this->booleanUtils->convert($this->getConfiguration()['timezone'])
        : false;

The only config values currently present in the code set the 'timezone'
config to FALSE using `<timezone>false</timezone>`. There are 0 occurances
of `<timezone>true</timezone>` in the code base at this time. This makes
me believe that the default value should not be FALSE but TRUE. Otherwise
the outcome will always be FALSE, be it explicit or by default.

This way the product review created_at times will be the same in the
catalog/product/edit adminhtml page as well as on the Marketing | Reviews
grid. They were not equal before this commit.

The same goes for the CMS page grid, the CMS block grid, the customer grid,
the online customers grid, and the sales grids.

I believe admins wint to see these times in their configured time zone and
not in UTC. Otherwise someone can appear to have placed an order in the
future which is very confusing.

Commit:
merge magento/develop into magento-nord/NORD-BUGFIXES-PR

Commit:
Cron uses the wrong timestamp method

Because the cron job uses a localized timestamp instead of a regular
timestamp, all times in the cron_schedule table are not the actual
times, but times with a double offset.

By using the regular timestamp, all times in cron_schedule are correct.
The cron job execution times are not affected by this change because
schedule generation and job execution depend on the same timestamp.
If both are in the same timezone all will go as intended.

Fixes #4237

ishakhsuvarov committed:
MAGETWO-69848: Fix #9924, prefill prefix and suffix in checkout shipping address #9925

 - Merge Pull Request magento/magento2#9925 from ajpevers/magento2:issue-9924
 - Merged commits:
   1. 6bf365feaf2186b8b9e4108e3e16921723b9d282
   2. 8b8c2fe49511e54efe401dba5a5e4d819253d075
   3. ff2319eab22cb5bfe887cf8d5da91641253ec46e
   4. 95b16840004d923272bec62ceaf3485d84143348
   5. 322c357b066d5fba2bcdf7276b29471e739a87d2

Commit:
Adapt unit test

slopukhov committed:
Merge branch 'MAGETWO-54066' of github.com:magento-troll/magento2ce into TrollPR

ishakhsuvarov committed:
MAGETWO-69848: Fix #9924, prefill prefix and suffix in checkout shipping address #9925

ishakhsuvarov committed:
MAGETWO-69849: Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69849-PR-9928

Conflicts:
	composer.lock

ishakhsuvarov committed:
MAGETWO-69849: upgrade tubalmartin/cssmin to 4.1.0 #9928

ishakhsuvarov committed:
MAGETWO-69849: upgrade tubalmartin/cssmin to 4.1.0 #9928

 - Upgraded expected minification result for the test

RomaKis committed:
Merge branch 'develop' into develop-pr4

dverkade committed:
- Removed abstract method
- Changed check to === null
- Added null to doc comments for $type parameter.

balex13 committed:
Merge pull request #1191 from magento-nord/NORD-BUGFIXES-PR

Fixed issues:
- MAGETWO-61560 Moving all attributes out from 'Advanced Pricing' group causes error
- MAGETWO-64221 [GitHub] Special price does not display correctly for future discounts #8375
- MAGETWO-69185 Unable to create shipping label if Region not specified for shipping origin

dverkade committed:
Added specific type hint for the tokenTypes parameter

Commit:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69515

vgoncharenko committed:
MAGETWO-67431: Functional test \Magento\ConfigurableProduct\Test\TestCase\CreateConfigurableProductEntityTest fails randomly

- fixed MAGETWO-69870
- fixed MAGETWO-57942
- fixed MAGETWO-69084
- fixed MAGETWO-60449

slopukhov committed:
Merge branch 'develop' of github.com:magento/magento2ce into TrollPR

vgoncharenko committed:
Merge branch 'develop' of https://github.com/magento/magento2ce into MAGETWO-67431

ishakhsuvarov committed:
MAGETWO-69879: Merge branch 'Fix_PaymentTokenFactory_interface_not_ending_class_name_with_Interface' of github.com:dverkade/magento2 into MAGETWO-69879-PR-9306

pdohogne-magento committed:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69648-UserModuleVersionCheck

Commit:
MAGETWO-69886: Cron uses the wrong timestamp method #9943

 - Merge Pull Request magento/magento2#9943 from ajpevers/magento2:issue-4237
 - Merged commits:
   1. 82f1bacaecffef5719333077ce8089a07a7c970b
   2. e080dde916feb4347fe6f93ed976a5f2655e8cff

Commit:
MAGETWO-69775: [Magento Cloud] Discount based off customer segments not applying to orders created in admin

rganin committed:
Merge pull request #1195 from magento-troll/TrollPR

MAGETWO-54066: [GITHUB] Invalid event prefix for catalog inventory #4857

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69879-PR-9306

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69868-PR-9902

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69849-PR-9928

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69848-PR-9925

Commit:
Merge remote-tracking branch 'mainline/develop' into MAGETWO-69832-PR-9915

Commit:
magento-engcom/magento2#25: Source API Implementation

-- add carriers links assignment

Commit:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69515

Commit:
MAGETWO-69879: Fix PaymentTokenFactory interface to have the "Interface" at the end of the name. #9306

Commit:
MAGETWO-69868: Fix static integrity classes tests in Windows #9902

Commit:
MAGETWO-69849: upgrade tubalmartin/cssmin to 4.1.0 #9928

Commit:
MAGETWO-69848: Fix #9924, prefill prefix and suffix in checkout shipping address #9925

Commit:
MAGETWO-69832: suggestion from #9338: add some command/option to the deploy command to refresh the version #9915

Commit:
MAGETWO-69229: Fix depends per group in system.xml #9391

Commit:
MAGETWO-69229: Fix depends per group in system.xml #9391

 - fixed code style

Commit:
Merge branch 'MAGETWO-69229-PR-9391' into develop-prs

Commit:
Merge pull request #1199 from magento-engcom/develop-prs

Public Pull Requests

magento/magento2#9306
magento/magento2#9902
magento/magento2#9928
magento/magento2#9925
magento/magento2#9915
magento/magento2#9391

Commit:
MAGETWO-69909: By default, show times in admin grids in the store timezone. #9941

 - Merge Pull Request magento/magento2#9941 from ajpevers/magento2:issue-9426
 - Merged commits:
   1. 049cc452247d7fd54d786b06c558231350137590

Commit:
MAGETWO-64996: Out of stock configurable product displays with zero price

Commit:
MAGETWO-69886: Cron uses the wrong timestamp method #9943

Commit:
MAGETWO-69909: By default, show times in admin grids in the store timezone. #9941

Commit:
Merge pull request #1201 from magento-engcom/develop-prs

Public Pull Requests

magento/magento2#9943
magento/magento2#9941

RomaKis committed:
Merge branch 'develop' into develop-pr4

Commit:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-69515

Commit:
MAGETWO-69515: Static files are deployed too slow for multiple locales

MomotenkoNatalia committed:
Merge pull request #1198 from magento-frontend/MAGETWO-69515

Fixed issue:
- MAGETWO-69515: Static files are deployed too slow for multiple locales

rganin committed:
Merge branch 'develop' of github.com:magento/magento2ce into MAGETWO-67431

rganin committed:
Merge pull request #1197 from magento-troll/MAGETWO-67431

MAGETWO-67431 Functional test \Magento\ConfigurableProduct\Test\TestCase\CreateConfigurableProductEntityTest fails randomly

Commit:
MAGETWO-64996: Out of stock configurable product displays with zero price

RomaKis committed:
Merge branch 'develop' into develop-pr4

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-69775-Discount' into Okapis-develop-pr

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-67438-Widget-Conditions' into Okapis-develop-pr

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-64996-Zero-Price-Out-Of-Stock' into Okapis-develop-pr

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-69648-UserModuleVersionCheck' into Okapis-develop-pr

cpartica committed:
Merge remote-tracking branch 'origin/MAGETWO-68976-Date-Format' into Okapis-develop-pr

cpartica committed:
Merge pull request #1204 from magento-okapis/Okapis-develop-pr

[Okapis] P1 P2

Commit:
magento-engcom/magento2#25: Source API Implementation

-- add carriers links assignment

Commit:
magento-engcom/magento2#25: Source API Implementation

-- fix tests

Commit:
magento-engcom/magento2#25: Source API Implementation

-- add carriers links assignment

Commit:
magento-engcom/magento2#25: Source API Implementation

-- add carriers links assignment

Commit:
Merge remote-tracking branch 'mainline/develop' into mni/source-management-ui

Commit:
magento-engcom/magento2#25: Source API Implementation

-- update composer.lock

naydav commented:
Internal build
https://bamboo.corp.magento.com/browse/M2-AT2096-13
Acceptance Tests plan is green.
